packages/core/src/actor/generic-conn-driver.ts

```diff
@@ -130,6 +128,27 @@ export function createGenericWebSocketDriver(
 
 			await promise;
 		},
+
+		getConnectionReadyState: (
+			_actor: AnyActorInstance,
+			conn: AnyConn,
+		): ConnectionReadyState | undefined => {
+			const ws = globalState.websockets.get(conn.id);
+			if (!ws) {
+				logger().warn("missing ws for getConnectionReadyState", {
+					connId: conn.id,
+				});
+				return undefined;
+			}
+
+			// InlineWebsocketAdapter2 does not update WSContext.readyState, it's always OPEN, so we need to check the raw WebSocket state.
+			const raw = ws.raw as WebSocket;
+			if (!raw) {
+				return ws.readyState as ConnectionReadyState;
+			}
+
+			return raw.readyState as ConnectionReadyState;
```

Comment:
my issue wasn't answered. why do we need to fall back to raw.readyState? why doesn't ws.readyState always work?